events/auditeventreporter.go

```diff
@@ -38,8 +39,7 @@ func (e *auditEventsReporterModule) ReportEvent(ctx context.Context, event Event
 		key = "DEFAULT-KEY"
 	}
 
-	msg := &sarama.ProducerMessage{Topic: e.topic, Key: sarama.StringEncoder(key), Value: sarama.StringEncoder(base64Event)}
-	_, _, err := e.producer.SendMessage(msg)
+	err := e.producer.SendMessageBytes([]byte(base64Event))
```

Comment:
Very important here, you should keep the partition key ! 

Then you should used SendPartitionedMessageBytes methods from  kafka-client